components/toast.js

```diff
@@ -0,0 +1,85 @@
+import { useRouter } from 'next/router'
+import React, { createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react'
+import Button from 'react-bootstrap/Button'
+import Toast from 'react-bootstrap/Toast'
+import ToastBody from 'react-bootstrap/ToastBody'
+import ToastContainer from 'react-bootstrap/ToastContainer'
+import styles from './toast.module.css'
+
+const ToastContext = createContext(() => {})
+
+let toastId = 0
+
+export const ToastProvider = ({ children }) => {
+  const router = useRouter()
+  const [toasts, setToasts] = useState([])
+  const dispatchToast = useCallback((toastConfig) => {
+    const id = toastId++
+    toastConfig = {
+      ...toastConfig,
+      id
+    }
+    if (toastConfig.autohide && toastConfig.delay > 0) {
+      setTimeout(() => {
```

Comment:
Not a big thing but why aren't we using the delay prop?

https://react-bootstrap.netlify.app/docs/components/toasts/#autohide

Reply:
I'll make the change myself. I'm just curious if I'm missing something

Reply:
I didn't want to leave the toast instance in the component tree after auto hide, so I chose to hide it by removing it from the array of toasts instead of using the delay prop.

Perhaps we could use the delay prop and also remove it from the array state? That might provide a transition out effect that would be nice. 

Idk if that made any sense...

Reply:
I think it makes sense. 

It looks like we can use the delay prop and `onClose` to remote it from the toasts array. I'll push that in a second.